src/utils/Table/TableContent.js

```diff
@@ -73,19 +75,44 @@ TableRow.propTypes = {
     RowActionsProps: PropTypes.object,
     RowContent: PropTypes.elementType.isRequired,
     RowCollapseComponent: PropTypes.elementType,
+    isSelectableTable: PropTypes.bool,
 };
 
 const TableContent = ({ rows, handleSelect, isRowSelected, RowActions, submitUndoableAction,
-    RowActionsProps, emptyMessage, numColumns, context, RowContent, RowCollapseComponent,
-}) => (
-    <TableBody>
-        {Object.keys(rows).length === 0 ?
+    RowActionsProps, emptyMessage, numColumns, context, RowContent, RowCollapseComponent, isSelectableTable, isLoading, error,
+}) => {
+    if (isLoading) {
+        const rowsArray = [0, 1, 2, 3, 4];
+        const iterableColArray = [];
+        for (let idx = 0; idx < numColumns + (isSelectableTable ? 1 : 0); idx++) {
+            iterableColArray.push({ key: idx });
+        }
+        return (
+            rowsArray.map((rowIdx) => (
+                <MUITableRow data-testid={`tableCellSkeleton-${rowIdx}`} key={rowIdx}>
+                    {iterableColArray.map((element) => (
+                        <TableCell
+                            key={element.key}
+                            align="right"
+                        >
+                            <Skeleton variant="rect" height={30} />
+                        </TableCell>
+                    ))}
+                </MUITableRow>
+
+            ))
+        );
```

Comment:
2 things: 
* You can pass the rowsPerPage to the TableContent (from base table) to know exactly how many rows you should mock
* In this case, since the skeletons don't change, it's safe to use the index of the array element as key, so the values do not matter, and the initialization is simpler

```suggestion
        const rowsMock = Array(rowsPerPage).fill(0);
        const columnsMock = Array(columns.length + (isSelectableTable ? 1 : 0)).fill(0);
        
        return (
            rowsMock.map((_, i) => (
                <MUITableRow data-testid={`tableCellSkeleton-${i}`} key={i}>
                    {columnsMock.map((_, j) => (
                        <TableCell
                            key={j}
                            align="right"
                        >
                            <Skeleton variant="rect" height={30} />
                        </TableCell>
                    ))}
                </MUITableRow>

            ))
        );
```